### DISCUSSION TITLE
How to handle many providers?

### DISCUSSION THREAD

**Discussion by seanoliver** [smol-ai/GodMode#41] [Ideas]
As of this writing, Menubar supports up to 4 chat providers—Bing, ChatGPT, Claude, and Bard. Each enabled provider sits in its own pane side-by-side. Following #32, we've made it easier to integrate additional providers, with plans already underway (see #35 to #40).

But as we add more providers, the side-by-side pane design might get overcrowded and unwieldy.

Here are some potential solutions we're considering:

1. Limit the active providers to 4 at any given time.
2. Allow users to adjust the provider limit to fit their screen size and needs.
3. Stack panes into rows, a maximum of 4 wide, though we'd need to think about a max cap.
4. Do nothing and let users manage their panes as they see fit.

Thought it would make sense to start this as a discussion rather than a ticket until we get closer to what we want the actual behavior to be.

Thoughts?

**Comment by swyxio** [comment]
i think we dont have to make a decision, users will naturally figure out how many they want, plus some freaks have ultrawide monitors lol
